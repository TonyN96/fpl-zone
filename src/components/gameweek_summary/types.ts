export interface SummaryStatType {
  label: string;
  teamCode?: number;
  playerName?: string;
  value?: string | number;
}
