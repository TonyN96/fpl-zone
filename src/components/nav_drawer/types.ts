export interface MenuItemType {
  id: string;
  label: string;
  type: "nav" | "auth";
  onClick: () => void;
}
