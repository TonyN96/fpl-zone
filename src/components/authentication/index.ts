export * from "./account/account_form";
export * from "./details_form";
export * from "./fpl_id_modal";
export * from "./login/fpl_id_login";
export * from "./login/login_form";
export * from "./register/register_form";
export * from "./reset/reset_form";
