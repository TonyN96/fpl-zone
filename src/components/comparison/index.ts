export * from "./add_players_to_comparison_modal";
export * from "./player_comparison";
export * from "./player_comparison_table";
export * from "./player_image_table_row";
export * from "./player_name_table_row";
export * from "./team_name_table_row";
