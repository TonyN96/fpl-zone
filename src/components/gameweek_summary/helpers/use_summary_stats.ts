import { useContext } from "react";
import { AppDataContext } from "app_content";
import { GetPlayerById, numberWithCommas } from "helpers";
import { AppData } from "types";

import { SummaryStatType } from "../gameweek_summary";

export const useSummaryStats = (): SummaryStatType[] => {
  const { gameweeks, players } = useContext(AppDataContext) as AppData;

  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const currentGameweek = gameweeks.find((gw) => gw.is_current)!;

  const starPlayerId = currentGameweek.top_element_info?.id;
  const starPlayer = starPlayerId ? GetPlayerById(starPlayerId, players) : null;

  const mostTransferredIn = currentGameweek.most_transferred_in
    ? GetPlayerById(currentGameweek.most_transferred_in, players)
    : null;

  const mostCaptained = currentGameweek.most_captained
    ? GetPlayerById(currentGameweek?.most_captained, players)
    : null;

  const mostViceCaptained = currentGameweek.most_vice_captained
    ? GetPlayerById(currentGameweek.most_vice_captained, players)
    : null;

  const mostTransferredInCount = mostTransferredIn
    ? numberWithCommas(mostTransferredIn?.transfers_in_event)
    : null;

  const starPlayerScore = currentGameweek.top_element_info
    ? `${currentGameweek.top_element_info?.points} pts`
    : "N/A";

  return ([
    {
      label: "highest score",
      value: `${currentGameweek.highest_score} pts`
    },
    {
      label: "average score",
      value: `${currentGameweek.average_entry_score} pts`
    },
    {
      label: "star player",
      teamCode: starPlayer?.team_code,
      playerName: starPlayer?.web_name || "",
      value: starPlayerScore
    },
    {
      label: "most captained",
      teamCode: mostCaptained?.team_code,
      playerName: mostCaptained ? mostCaptained.web_name : "N/A"
    },
    {
      label: "most vice-captained",
      teamCode: mostViceCaptained?.team_code,
      playerName: mostViceCaptained ? mostViceCaptained.web_name : "N/A"
    },
    {
      label: "most transferred in",
      teamCode: mostTransferredIn?.team_code,
      playerName: mostTransferredIn?.web_name || "",
      value: mostTransferredInCount || ""
    }
  ]);
};
