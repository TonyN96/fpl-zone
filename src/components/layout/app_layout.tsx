import React from "react";
import NavDrawer from "components/nav_drawer/nav_drawer";
import { Box, Container, Typography } from "@mui/material";

interface LayoutProps {
  activeLabel: string;
  direction: string;
}

export default function AppLayout(props: React.PropsWithChildren<LayoutProps>): JSX.Element {
  return (
    <Box component="div" sx={{ display: "flex" }}>
      <NavDrawer activeLabel={props.activeLabel} />
      <Box
        component="main"
        sx={{
          flexGrow: 1,
          height: "100vh",
          maxHeight: "100vh",
          display: "flex",
          flexDirection: "column",
        }}
      >
        <Typography
          textAlign="center"
          component="h1"
          variant="h3"
          sx={{ mt: 3, mb: 2, height: "auto" }}
        >
          {props.activeLabel.toUpperCase()}
        </Typography>
        <Container sx={{ flexGrow: "1", mb: 3 }} maxWidth="xl">
          <Box
            sx={{
              height: "100%",
              maxHeight: "100vh",
              display: "flex",
              flexDirection: props.direction,
              justifyContent: "space-between",
              columnGap: 3,
              rowGap: 3,
            }}
          >
            {props.children}
          </Box>
        </Container>
      </Box>
    </Box>
  );
}
