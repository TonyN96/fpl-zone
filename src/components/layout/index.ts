export * from "./app_layout";
export * from "./component_container";
export * from "./notifier";
export * from "./startup";
