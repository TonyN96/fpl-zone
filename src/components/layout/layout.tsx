import React from "react";
import NavDrawer from "components/nav_drawer/nav_drawer";
import { Box } from "@mui/material";

interface LayoutProps {
  active: string;
}

export default function Layout(props: React.PropsWithChildren<LayoutProps>) {
  return (
    <Box component="div" sx={{ display: "flex" }}>
      <NavDrawer activeId={props.active} />
      <Box component="main" sx={{ flexGrow: 1 }}>
        {props.children}
      </Box>
    </Box>
  );
}
