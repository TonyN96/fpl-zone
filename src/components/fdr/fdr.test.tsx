import React from "react";
import { render, screen, waitFor } from "@testing-library/react";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import { getLocalImage } from "helpers";
import { mockFixtures, mockPlayers, mockTeams } from "test";
import { MockProviders } from "test/mock_providers";
import { Player } from "types";

import FdrTable from "components/fdr/fdr";
import { notifierMessageMap as msgMap } from "components/layout";

import "@testing-library/jest-dom/extend-expect";

let mockGameweekIds = [11, 12, 13, 14, 15];

jest.mock("hooks/use_next_five_gameweek_ids", () => ({
  useNextFiveGameweekIds: (): number[] => mockGameweekIds
}));

describe("FDR Tests", () => {
  let mockFdrPlayers: Player[];

  const mockAdapter = new MockAdapter(axios);
  const mockReturnedFixtures = [mockFixtures.slice(0, 3)];

  mockAdapter.onGet().reply(200, mockReturnedFixtures);

  const createComponent = (
    playerFdr = false,
    gameweekIds = [11, 12, 13, 14, 15]
  ): JSX.Element => {
    mockGameweekIds = gameweekIds;

    return (
      <MockProviders>
        <FdrTable players={playerFdr ? mockPlayers : undefined} />
      </MockProviders>
    );
  };

  it("displays gameweek numbers correctly", async () => {
    render(createComponent());

    await screen.findByTestId("fdr-container").then(() => {
      const columnTitlesRow = screen.getByTestId("table-head-column-title");

      mockGameweekIds.forEach((id) => {
        expect(columnTitlesRow).toHaveTextContent(`GW ${id}`);
      });
    });
  });

  describe("Team FDR", () => {
    it("displays team names correctly", async () => {
      render(createComponent());

      await screen.findByTestId("fdr-container").then(() => {
        mockTeams.forEach((team) => {
          const teamName = screen.getByTestId(`base-item-text-${team.id}`);

          expect(teamName).toHaveTextContent(team.name);
        });
      });
    });

    it("displays team crest images correctly", async () => {
      render(createComponent());

      await screen.findByTestId("fdr-container").then(() => {
        mockTeams.forEach((team) => {
          const img = screen.getByTestId(`base-item-crest-img-${team.id}`);
          const imgUrl = getLocalImage(`crests/${team.code}`);

          expect(img).toHaveAttribute("src", imgUrl);
        });
      });
    });
  });

  describe("Player FDR", () => {
    mockFdrPlayers = mockPlayers.slice(0, 14);

    it("Displays player names correctly", async () => {
      render(createComponent(true));

      await screen.findByTestId("fdr-container").then(() => {
        mockFdrPlayers.forEach((player) => {
          const playerName = screen.getByTestId(`base-item-text-${player.id}`);

          expect(playerName).toHaveTextContent(player.web_name);
        });
      });
    });

    it("displays team crest images correctly", async () => {
      render(createComponent(true));

      await screen.findByTestId("fdr-container").then(() => {
        mockFdrPlayers.forEach((player) => {
          const img = screen.getByTestId(`base-item-crest-img-${player.id}`);
          const imgUrl = getLocalImage(`crests/${player.team_code}`);

          expect(img).toHaveAttribute("src", imgUrl);
        });
      });
    });
  });

  describe("Notifier", () => {
    it("displays expected message if season has finished", async () => {
      render(createComponent(false, []));

      await screen.findByTestId("notifier-container").then(() => {
        const notifierContainer = screen.getByTestId("notifier-container");
        expect(notifierContainer).toHaveTextContent(msgMap.seasonFinished);
      });
    });

    it("displays expected message if fixtures are being fetched", async () => {
      render(createComponent());

      await waitFor(() => {
        const notifierContainer = screen.getByTestId("notifier-container");
        expect(notifierContainer).toHaveTextContent("Fetching fixture data..");
      });
    });
  });
});
