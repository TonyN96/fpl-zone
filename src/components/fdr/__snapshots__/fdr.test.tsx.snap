// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FDR Tests FDR table for players Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-16r1tk9"
  >
    <div
      class="MuiBox-root css-icwvrz"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Fetching fixture data..
      </p>
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1lmcs20-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FDR Tests FDR table for teams Snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-16r1tk9"
  >
    <div
      class="MuiBox-root css-icwvrz"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Fetching fixture data..
      </p>
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1lmcs20-MuiCircularProgress-root"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
